testcases/InstallHostOS.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/python2
+# OpenPOWER Automated Test Project
+#
+# Contributors Listed Below - COPYRIGHT 2018
+# [+] International Business Machines Corp.
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+
+
+import unittest
+import time
+import subprocess
+
+import OpTestConfiguration
+from common.OpTestUtil import OpTestUtil
+from common.OpTestSystem import OpSystemState
+from common.Exceptions import CommandFailed
+from common import OpTestInstallUtil
+
+
+class InstallHostOS(unittest.TestCase):
+    def setUp(self):
+        self.conf = OpTestConfiguration.conf
+        self.host = self.conf.host()
+        self.ipmi = self.conf.ipmi()
+        self.system = self.conf.system()
+        self.bmc = self.conf.bmc()
+        self.util = OpTestUtil()
+        self.bmc_type = self.conf.args.bmc_type
+        if not (self.conf.args.os_repo or self.conf.args.os_cdrom):
+            self.fail("Provide installation media for installation, --os-repo is missing")
+        if not (self.conf.args.host_gateway and self.conf.args.host_dns and self.conf.args.host_submask):
+            self.fail("Provide host network details refer, --host-{gateway,dns,submask}")
+        if not self.conf.args.host_scratch_disk:
+            self.fail("Provide proper host disk to install refer, --host-scratch-disk")
+
+    def runTest(self):
+        self.system.goto_state(OpSystemState.PETITBOOT_SHELL)
+        self.c = self.system.sys_get_ipmi_console()
+        self.system.host_console_unique_prompt()
+
+        # Set the install paths
+        base_path = "osimages/hostos"
+        boot_path = "ppc/ppc64"
+        vmlinux = "vmlinuz"
+        initrd = "initrd.img"
+        ks = "hostos.ks"
+
+        retry = 30
+        while retry > 0:
+            try:
+                self.c.run_command("ifconfig -a")
+                break
+            except CommandFailed as cf:
+                if cf.exitcode is 1:
+                    time.sleep(1)
+                    retry = retry - 1
+                    pass
+                else:
+                    raise cf
+        my_ip = self.system.get_my_ip_from_host_perspective()
+        if not my_ip:
+            self.fail("unable to get the ip from host")
+        self.system.host_console_unique_prompt()
```

Comment:
We can remove this as one time at start up is fine.,

Reply:
it was failing without it, might be due to `system.get_my_ip_from_host_perspective()` , not sure though...